File: Baekjoon/Greedy/13305.py
# https://www.acmicpc.net/problem/13305

import sys
input = sys.stdin.readline

k = int(input())
dist = list(map(int, input().split()))
cost = list(map(int, input().split()))
min_cost = 0

min_pos = cost[0]
for i in range(k - 1):
    if min_pos > cost[i]:
        min_pos = cost[i]
    min_cost += min_pos * dist[i]

print(min_cost)

File: Baekjoon/Greedy/13305.cpp
// https://www.acmicpc.net/problem/13305

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<long long> dist(n - 1);
    for (int i = 0; i < n - 1; i++) {
        cin >> dist[i];
    }

    vector<long long> cost(n);
    for (int i = 0; i < n; i++) {
        cin >> cost[i];
    }

    long long ans = 0;
    long long minCost = cost[0];
    for (int i = 0; i < n - 1; i++) {
        if (cost[i] < minCost) {
            minCost = cost[i];
        }
        ans += minCost * dist[i];
    }

    cout << ans << '\n';

    return 0;
}